### PR TITLE
Handle use of genericArgumentFactories when generic type is T? in field usage

### PR DESCRIPTION
**Pull request by kevmoo**
Fixes https://github.com/google/json_serializable.dart/issues/803 (for real)
Closes https://github.com/google/json_serializable.dart/pull/837

### CODE REVIEW

**Comment by kevmoo** [comment]
CC @sooxt98

**Comment by sooxt98** [comment]
@kevmoo wow that's alot of changes. gonna try it out later

File: json_serializable/test/generic_files/generic_argument_factories_nullable.dart [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
```
Comment (natebosch) [inline json_serializable/test/generic_files/generic_argument_factories_nullable.dart:1]:
```suggestion
// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
```

**Review by natebosch — APPROVED** [review]
Can you please edit the commit message before merging - it could use more detail and a shorter title line.

**Comment by sooxt98** [comment]
@kevmoo yay done testing, it works like expected; 🎉

**Comment by sooxt98** [comment]
@kevmoo yay done testing, it works like expected; 🎉

**Pull request merged**